persistence/test/test.go

```diff
@@ -66,5 +80,24 @@ func ResetPGTestDB(t *testing.T, settings *TestDBSettings) (*pg.DB, string) {
 
 	require.Equal(t, settings.ExpectedSchemaVersion, newVersion)
 
-	return db, dsn
+	db.Close()
+
+	return dbSettings
+}
+
+func DropTestDB(t *testing.T, opts pg.Options) {
+	dbName := opts.Database
+
+	// To drop the DB, we need a connection to the server that isn't
+	// accessing the DB itself. We connect to the template1 DB, which
+	// always exists.
+	opts.Database = "template1"
+
+	template1 := pg.Connect(&opts)
```

Comment:
Doesn't this need the same lock again?

It does make me wonder if connecting to `template1` is such a good idea.

Are you sure `postgres` isn't also available generally? That is what people on the internet seem to recommend.

Reply:
Per discussion, we removed the lock altogether, replaced `template1` with `postgres` (which should work in most situations), and added some context from the caller package to also remove conflicts between tests from different packages running in separate binaries.